driver-async/src/main/com/mongodb/async/client/MongoClients.java

```diff
@@ -218,25 +224,57 @@ public static CodecRegistry getDefaultCodecRegistry() {
         return com.mongodb.MongoClientSettings.getDefaultCodecRegistry();
     }
 
-    @SuppressWarnings("deprecation")
-    private static StreamFactory getStreamFactory(final MongoClientSettings settings, final String streamType,
-                                                  final boolean isHeartbeat) {
+
+    private static MongoClient createWithTlsChannel(final MongoClientSettings settings,
+                                                    @Nullable final MongoDriverInformation mongoDriverInformation) {
+        if (!isJava8()) {
+            throw new MongoClientException("TLS is only supported natively with Java 8 and above. Please use Netty instead");
+        }
+        final TlsChannelStreamFactoryFactory streamFactoryFactory = new TlsChannelStreamFactoryFactory();
+        StreamFactory streamFactory = streamFactoryFactory.create(settings.getSocketSettings(), settings.getSslSettings());
+        StreamFactory heartbeatStreamFactory = streamFactoryFactory.create(settings.getHeartbeatSocketSettings(),
+                settings.getSslSettings());
+        return createMongoClient(settings, mongoDriverInformation, streamFactory, heartbeatStreamFactory,
+                new Closeable() {
+                    @Override
+                    public void close() {
+                        streamFactoryFactory.close();
```

Comment:
Is it worth adding close to the StreamFactoryFactory interface? - then all we need do for the different types is return the StreamFactoryFactory. (Both TLSChannel and Netty have a close method).

Reply:
Adding a new method to this interface is a breaking change for any implementers of the interface.  Maybe do this in 4.0?